cmd/mirror-main.go

```diff
@@ -493,25 +488,15 @@ exit:
 
 // when using a struct for copying, we could save a lot of passing of variables
 func (mj *mirrorJob) mirror(ctx context.Context, cancelMirror context.CancelFunc) {
-	if globalQuiet || globalJSON {
-	} else {
-		// Enable progress bar reader only during default mode
-		mj.scanBar = scanBarFactory()
-	}
-
 	// start the status go routine
 	mj.startStatus()
 
-	// Starts additional watcher thread for watching for new events.
-	if mj.isWatch {
-		go mj.watchMirror(ctx, cancelMirror)
```

Comment:
It is not commented in the code, but the reason why we have this in a go-routine is to avoid missing new changes in source such as removing files, adding new files.. this is very useful especially when mirroring takes some times. Any reason why you are removing this @harshavardhana ?

Reply:
I know the reason @vadmeste , i generally thought it is not a big deal. Since they will be caught up later but i see now that it might be not a good idea and it is possible that we do not exit properly.